v3/src/components/graph/models/graph-layout.ts

```diff
@@ -63,18 +64,18 @@ export class GraphLayout {
       // the plot. We tried work arounds to get gridlines that were _not_ part of the axis element with the result
       // that the gridlines got out of synch with axis tick marks during drag. So we have this inelegant solution
       // that shouldn't affect the top and right axes when we get them but
```

Comment:
I did try to find conditions where the calculations that I removed were still necessary, but could not find them.  If I have missed something, or omitted a calculation that anticipated future changes I can put them back in.  I found it easier to reason about by just including the necessary calculations and separating them by axis.  I imagine that if we were to need to add the other calculations down the line it might look more like this: 

```ts
      if (place === "bottom"){
        newBounds.top = this.plotHeight
        newBounds.height = // some calculation
        newBounds.width = // some calculation
      }
```

I see that the width of the axes needs to stay flexible, so for that reason intentionally let the y axis drop zone overlap with the table next door when the words are long enough.  

Anyway, let me know what you think.

Reply:
I'll add that each of these are basically doing the same thing, re-limiting the size of the dropzone so that it does not follow the size of the svg when that svg extends into the plot because of gridlines.  In the first case, it is not always required, but always valid.  In the second case, it is only valid when the gridlines are on.

Reply:
@bfinzer thanks for pointing that out, I should have noticed the legends were mentioned.  I restored them within the new structure.

Reply:
>  restrict the drop zone to the graph component in all cases
OK I made a change:

```ts
if (place === "left"){
        newBounds.height = Math.min(bounds.height, this.graphHeight - this.legendHeight)
        newBounds.left = 0
        newBounds.width = this.graphWidth - this.plotWidth - this.margin.left - this.margin.right
        // if gridlines are present, axis will grow to .width + plotWidth, so we recalculate
        if (bounds.width > this.plotWidth){
          newBounds.width -= this.plotWidth
        }
      }
```